seed/csharp-sdk/imdb/include-exception-handler/src/SeedApi/Imdb/ImdbClient.cs

```diff
@@ -0,0 +1,135 @@
+            {
+                var responseBody = await response.Raw.Content.ReadAsStringAsync();
+                try
+                {
+                    return JsonUtils.Deserialize<string>(responseBody)!;
+                }
+                catch (JsonException e)
+                {
+                    throw new SeedApiException("Failed to deserialize response", e);
+                }
+            }
+
+            {
+                var responseBody = await response.Raw.Content.ReadAsStringAsync();
+                throw new SeedApiApiException(
+                    $"Error with status code {response.StatusCode}",
+                    response.StatusCode,
+                    responseBody
+                );
+            }
+        });
+    }
+
+    /// <example>
+    /// <code>
+    /// await client.Imdb.GetMovieAsync("movieId");
+    /// </code>
+    /// </example>
+    public async Task<Movie> GetMovieAsync(
+        string movieId,
+        RequestOptions? options = null,
+        CancellationToken cancellationToken = default
+    )
+    {
+        return await _exceptionHandler.TryCatchAsync(async () =>
+        {
+            var response = await _client
+                .SendRequestAsync(
+                    new RawClient.JsonApiRequest
+                    {
+                        BaseUrl = _client.Options.BaseUrl,
+                        Method = HttpMethod.Get,
+                        Path = $"/movies/{JsonUtils.SerializeAsString(movieId)}",
+                        Options = options,
+                    },
+                    cancellationToken
+                )
+                .ConfigureAwait(false);
+            if (response.StatusCode is >= 200 and < 400)
+            {
+                var responseBody = await response.Raw.Content.ReadAsStringAsync();
+                try
+                {
+                    return JsonUtils.Deserialize<Movie>(responseBody)!;
+                }
+                catch (JsonException e)
+                {
+                    throw new SeedApiException("Failed to deserialize response", e);
+                }
+            }
+
+            {
+                var responseBody = await response.Raw.Content.ReadAsStringAsync();
+                try
+                {
+                    switch (response.StatusCode)
+                    {
+                        case 404:
+                            throw new MovieDoesNotExistError(
+                                JsonUtils.Deserialize<string>(responseBody)
+                            );
+                    }
+                }
+                catch (JsonException)
+                {
+                    // unable to map error response, throwing generic error
+                }
+                throw new SeedApiApiException(
+                    $"Error with status code {response.StatusCode}",
+                    response.StatusCode,
+                    responseBody
+                );
+            }
+        });
```

Comment:
If we're `await`'ing anything, we should always add `ConfigureAwait(false)`

Reply:
Great catch - done!